src/VernaContextProvider.tsx

```diff
@@ -71,18 +84,76 @@ function VernaContextProvider({
   defaultUiSchema,
   defaultWidget,
   defaultReadOnly,
+  defaultSelector,
   children,
 }: PropsWithChildren<VernaContextProviderProps>) {
-  const [schema, setSchema] = useState<JSONSchema7>(defaultSchema);
-  const [uiSchema, setUiSchema] = useState<UiSchema>(
-    genDefaultUiSchema(defaultSchema, defaultUiSchema),
+  // The full schemas are not used by the lib itself but may be used for further implementation
+  const [fullSchema, setFullSchema] = useState<JSONSchema7>(defaultSchema || {});
+  const [fullUiSchema, setFullUiSchema] = useState<UiSchema>(
+    genDefaultUiSchema(defaultSchema || {}, defaultUiSchema || {}),
   );
```

Comment:
`defaultSchema` and `defaultUiSchema` have already a default value.
```suggestion
  // The full schemas are not used by the lib itself but may be used for further implementation
  const [fullSchema, setFullSchema] = useState<JSONSchema7>(defaultSchema);
  const [fullUiSchema, setFullUiSchema] = useState<UiSchema>(
    genDefaultUiSchema(defaultSchema, defaultUiSchema),
  );
```